src/crypto/crypto_context.cc

```diff
@@ -61,18 +61,16 @@ inline X509_STORE* GetOrCreateRootCertStore() {
 // Takes a string or buffer and loads it into a BIO.
 // Caller responsible for BIO_free_all-ing the returned object.
 BIOPointer LoadBIO(Environment* env, Local<Value> v) {
-  HandleScope scope(env->isolate());
-
-  if (v->IsString()) {
-    Utf8Value s(env->isolate(), v);
-    return NodeBIO::NewFixed(*s, s.length());
-  }
-
-  if (v->IsArrayBufferView()) {
-    ArrayBufferViewContents<char> buf(v.As<ArrayBufferView>());
-    return NodeBIO::NewFixed(buf.data(), buf.length());
+  if (v->IsString() || v->IsArrayBufferView()) {
+    BIOPointer bio(BIO_new(BIO_s_secmem()));
+    if (!bio)
+      return nullptr;
+    ByteSource bsrc = ByteSource::FromStringOrBuffer(env, v);
+    int written = BIO_write(bio.get(), bsrc.data<char>(), bsrc.size());
+    if (written != bsrc.size())
+      return nullptr;
```

Comment:
To avoid compiler warnings and/or wraparound, this should look something like this:
```suggestion
    if (bsrc.size() > INT_MAX)
      return nullptr;
    int written = BIO_write(bio.get(), bsrc.data<char>(), bsrc.size());
    if (written < 0)
      return nullptr;
    if (static_cast<size_t>(written) != bsrc.size())
      return nullptr;
```
The greater-than-INT_MAX case is unlikely but not impossible. Arraybuffers can be > 2 GB.